arangod/RocksDBEngine/RocksDBMetaCollection.cpp

```diff
@@ -478,10 +459,16 @@ rocksdb::SequenceNumber RocksDBMetaCollection::serializeRevisionTree(
 
 Result RocksDBMetaCollection::rebuildRevisionTree() {
   std::unique_lock<std::mutex> guard(_revisionTreeLock);
-  _revisionTree = std::make_unique<containers::RevisionTree>(
-      6, _logicalCollection.minRevision().id());
-
-  Result res = basics::catchToResult([this]() -> Result {
+  struct {
+    std::unique_ptr<containers::RevisionTree> tree;
+    std::atomic<rocksdb::SequenceNumber> applied;
```

Comment:
As far as I can see, the `applied` member variable is only updated inside this function for a function-local variable. So there should be no other threads modifying it at the same time. Just wondering why the variable is declared atomic here. It would probably be ok to declare it non-atomic, i.e.
```suggestion
    rocksdb::SequenceNumber applied;
```
and then of course adjust the 2 or 3 places that store or load its value.